### ISSUE TITLE
Allow remote inbox notifications to link to other pages in wp-admin

### ISSUE THREAD

**Issue by andfinally** [woothemes/woocommerce#32129]
@budzanowski had a requirement to link from a Remote Inbox Notification to a wp-admin page outside of WooCommerce, `wp-admin/plugins.php`. But looking in `SpecRunner::run_spec`, he found that any URL with `url_is_admin_query` in the spec is passed through `wc_admin_url`, which means that we can only target WooCommerce URLs like `admin.php?page=wc-admin' . $path`.

It would be great if the notification engine could support linking to local wp-admin URLs outside of WooCommerce.

[Slack](p1647510260578519-slack-C4E0DVDB2)

### Implementation tasks

- [ ] How to best handle that `url_is_admin_query` is poorly named (it ambiguously refers to WCA queries, not "wp-admin", as might be expected)?
- [ ] Write tests

**Comment by joshuatf** [comment]
Note that @joelclimbsthings has an open PR ( https://github.com/woocommerce/woocommerce/pull/33124 ) that addresses how links are handled for various URL types.  There might be an opportunity for combining the efforts there and here.

**Comment by louwie17** [comment]
The changes for this have been merged, so this will be available in WooCommerce 6.7.
If you do add a notification with an `wp-admin` link make sure to add this rule:
```
{
"type": "plugin_version",
"plugin": "woocommerce",
"operator": ">=",
"version": "6.7"
}
```
So that the note won't be displayed on older versions where the link would be broken.
